zuul/connection/github.py

```diff
@@ -473,45 +475,48 @@ def getPullBySha(self, sha):
         if len(pulls) > 1:
             raise Exception('Multiple pulls found with head sha %s' % sha)
 
-        log_rate_limit(self.log, self.github)
+        log_rate_limit(self.log, github)
         if len(pulls) == 0:
             return None
         return pulls.pop()
 
     def getPullFileNames(self, owner, project, number):
+        github = self.getGithubClient()
         filenames = [f.filename for f in
-                     self.github.pull_request(owner, project, number).files()]
-        log_rate_limit(self.log, self.github)
+                     github.pull_request(owner, project, number).files()]
+        log_rate_limit(self.log, github)
         return filenames
 
     def getPullReviews(self, owner, project, number):
         # make a list out of the reviews so that we complete our
         # API transaction
+        github = self.getGithubClient()
         reviews = [review.as_dict() for review in
-                   self.github.pull_request(owner, project, number).reviews()]
-        log_rate_limit(self.log, self.github)
+                   github.pull_request(owner, project, number).reviews()]
+        log_rate_limit(self.log, github)
         return reviews
 
     def getUser(self, login):
-        return GithubUser(self.github, login)
+        return GithubUser(self.getGithubClient(), login)
 
     def getUserUri(self, login):
         return 'https://%s/%s' % (self.git_host, login)
 
     def getRepoPermission(self, owner, project, login):
         # This gets around a missing API call
         # need preview header
+        github = self.getGithubClient()
```

Comment:
Shouldn't this be done with the context of the owner/project so that the installation key is used?